src/transformers/modeling_utils.py

```diff
@@ -739,17 +739,30 @@ def load(module: nn.Module, prefix=""):
 
                 missing_keys.extend(head_model_state_dict_without_base_prefix - base_model_state_dict)
 
+            if len(unexpected_keys) > 0:
+                logger.warning(
+                    f"Some weights of the model checkpoint at {pretrained_model_name_or_path} were not used when "
+                    f"initializing {model.__class__.__name__}: {unexpected_keys}\n"
+                    f"- This IS expected if you are initializing {model.__class__.__name__} from the checkpoint of a model trained on another task "
+                    f"or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPretraining model).\n"
+                    f"- This IS NOT expected if you are initializing {model.__class__.__name__} from the checkpoint of a model that you expect "
+                    f"to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model)."
+                )
+            else:
+                logger.warning(
```

Comment:
```suggestion
                logger.info(
```

I think "standard operation" cases should not pop a warning, just an info

Reply:
Otherwise it gets a bit noisy in downstream use cases